src/components/Footer.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useContext } from 'react';
+
+import classNames from 'classnames';
+import { TodoContext } from '../services/TodoContext';
+import { Filter } from '../types/Filter';
+
+export const Footer: React.FC = () => {
+  const { todos, filter, deleteTodo, setFilter } = useContext(TodoContext);
+
+  const completedTodos = todos.filter(todo => todo.completed);
+  const activeTodos = todos.filter(todo => !todo.completed);
+
+  const handleFilterAll = () => {
+    setFilter(Filter.All);
+  };
+
+  const handleFilterActive = () => {
+    setFilter(Filter.Active);
+  };
+
+  const handleFilterCompleted = () => {
+    setFilter(Filter.Completed);
+  };
```

Comment:
Instead 3 handler use closing
```suggestion
  const onFilterChange = (newFilter: Filter) => () =>{
    setFilter(newFilter);
  };
```